### DISCUSSION TITLE
SolidOS docker version

### DISCUSSION THREAD

**Discussion by gp125fi** [solid/solidos#112] [Archive] [answered]
I would like to know how to run SolidOS via docker. I wonder if there is any SolidOS docker version like docker image, dockerfile, docker documentation, etc.

**Comment by angelo-v** [comment, marked as answer]
What do you mean by a docker version? SolidOS is a web app, that is usually deployed on a Solid Server, it e.g. comes as a default frontend for NSS. There is already a docker version of NSS, see https://hub.docker.com/r/nodesolidserver/node-solid-server, does this help? If not, what are you trying to achieve?

**Reply by gp125fi** [reply]
Docker version of NSS with SolidOS is too heavy. I would like to separate frontend and backend from monolithic application. If there is a SolidOS docker version without any backend service, monolith could gradually become microservice. So the question may become how to separate SolidOS from NSS to make a pure web app?

**Reply by angelo-v** [reply]
SolidOS is a web app, so it has to be served by some kind of web server, even when dockerized. One could use a docker container with nginx or apache serving it, but this only would make sense for the "browse.html" variant ([as deployed here](https://solid.github.io/mashlib/dist/browse.html)), as it has an URL bar to navigate to a specific resource. The default SolidOS always displays data from the current URL, which only makes sense when served from a Solid Server.

Perhaps you can tell more about your use-case? If you e.g. want to run SolidOS on desktop you could also use [data kitchen](https://github.com/solid/data-kitchen)

**Comment by bourgeoa** [comment]
You can run SolidOS as a web app either from :
- https://solid.github.io/mashlib/dist/browse.html
- or any NSS pod https://anypod/browse.html

**Comment by gp125fi** [comment]
The information is useful. Thank you all for reply.
